modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ServiceTests.java

```diff
@@ -54,11 +72,11 @@ public void testCachedClientsAreReleased() throws IOException {
         assertSame(clientSettings, otherClientSettings);
         final AmazonS3Reference reference = s3Service.client(metadata1);
         reference.close();
-        s3Service.doClose();
+        s3Service.onBlobStoreClose();
```

Comment:
This test was not doing what it meant to do since we changed the S3Service to be a lifecyle component. When `doClose` is called, the lifecycle should be closed and no longer allow creating new clients. So we need to change the call to  `onBlobStoreClose`.